File: JavaScript/BrickWall/BrickWall.js
/**
 * (276ms)
 * @param {number[][]} wall
 * @return {number}
 */
var leastBricks = function(wall) {
    var count = new Map(), max = 0;
    for(var row of wall) {
    	var now = 0;
    	for(var i = 0; i < row.length - 1; i++) {
    		now += row[i];
    		if(count.has(now)) {
    			count.set(now, count.get(now) + 1);
    		} else {
    			count.set(now, 1);
    		}
    		max = Math.max(max, count.get(now));
    	}
    	console.log(count);
    }
    return wall.length - max;
};

File: Python/BrickWall/BrickWall.py
class Solution(object):
	# (92ms)
    def leastBricks(self, wall):
        """
        :type wall: List[List[int]]
        :rtype: int
        """
        dt,maxCount = {}, 0
        for row in wall:
        	now = 0
        	for width in row[:-1]:
        		now += width
        		dt[now] = dt.get(now, 0) + 1
        		maxCount = max(maxCount, dt[now])
        return len(wall) - maxCount

print Solution().leastBricks([[1,2,2,1], [3,1,2], [1,3,2], [2,4], [3,1,2], [1,3,1,1]])